src/pretix/base/datasync/sourcefields.py

```diff
@@ -0,0 +1,548 @@
+
+from django.db.models import Max
+from django.utils.translation import gettext_lazy as _
+
+from pretix.base.models import Checkin, InvoiceAddress, Order, Question
+from pretix.base.settings import PERSON_NAME_SCHEMES
+
+
+def get_answer(op, question_identifier=None):
+    a = None
+    if op.addon_to:
+        if "answers" in getattr(op.addon_to, "_prefetched_objects_cache", {}):
+            try:
+                a = [
+                    a
+                    for a in op.addon_to.answers.all()
+                    if a.question.identifier == question_identifier
+                ][0]
+            except IndexError:
+                pass
+        else:
+            a = op.addon_to.answers.filter(
+                question__identifier=question_identifier
+            ).first()
+
+    if "answers" in getattr(op, "_prefetched_objects_cache", {}):
+        try:
+            a = [
+                a
+                for a in op.answers.all()
+                if a.question.identifier == question_identifier
+            ][0]
+        except IndexError:
+            pass
+    else:
+        a = op.answers.filter(question__identifier=question_identifier).first()
+
+    if not a:
+        return ""
+    else:
+        if a.question.type in (Question.TYPE_CHOICE, Question.TYPE_CHOICE_MULTIPLE):
+            return [str(o.identifier) for o in a.options.all()]
+        if a.question.type == Question.TYPE_BOOLEAN:
+            return a.answer == "True"
+        return a.answer
+
+
+def get_payment_date(order):
+    if order.status == Order.STATUS_PENDING:
+        return None
+
+    return isoformat_or_none(order.payments.aggregate(m=Max("payment_date"))["m"])
+
+
+def isoformat_or_none(dt):
+    return dt and dt.isoformat()
+
+
+def first_checkin_on_list(list_pk, position):
+    checkin = position.checkins.filter(
+        list__pk=list_pk, type=Checkin.TYPE_ENTRY
+    ).first()
+    if checkin:
+        return isoformat_or_none(checkin.datetime)
+
+
+def split_name_on_last_space(name, part):
+    name_parts = name.rsplit(" ", 1)
+    return name_parts[part] if len(name_parts) > part else ""
+
+
+ORDER_POSITION = 'position'
+ORDER = 'order'
+EVENT = 'event'
+EVENT_OR_SUBEVENT = 'event_or_subevent'
+AVAILABLE_MODELS = {
+    'OrderPosition': (ORDER_POSITION, ORDER, EVENT_OR_SUBEVENT, EVENT),
+    'Order': (ORDER, EVENT),
+}
+
+
+DataFieldInfo = namedtuple(
+    'DataFieldInfo',
+    field_names=('required_input', 'key', 'label', 'type', 'enum_opts', 'getter', 'deprecated'),
+    defaults=[False]
+)
+
+
+def get_invoice_address_or_empty(order):
+    try:
+        return order.invoice_address
+    except InvoiceAddress.DoesNotExist:
+        return InvoiceAddress()
+
+
+def get_data_fields(event, for_model=None):
+    """
+    Returns tuple of (required_input, key, label, type, enum_opts, getter)
+
+    type is one of the hubspot data types as specified in
+    https://developers.hubspot.com/docs/api/crm/properties#property-type-and-fieldtype-values
```

Comment:
We probably do not want to reference hubspot here any more. What types do we use instead?

Reply:
ah... I switched to pretix Question types

```suggestion
    Type is one of the pretix Question types as defined in Question.TYPE_CHOICES.
```